packages/react-components/react-provider/library/src/components/FluentProvider/FluentProvider.tsx

```diff
@@ -1,17 +1,23 @@
 import * as React from 'react';
+
+import type { ForwardRefComponent } from '@fluentui/react-utilities';
+
 import { renderFluentProvider_unstable } from './renderFluentProvider';
 import { useFluentProvider_unstable } from './useFluentProvider';
 import { useFluentProviderStyles_unstable } from './useFluentProviderStyles.styles';
 import { useFluentProviderContextValues_unstable } from './useFluentProviderContextValues';
 import type { FluentProviderProps } from './FluentProvider.types';
 
-export const FluentProvider = React.forwardRef<HTMLElement, FluentProviderProps>((props, ref) => {
+export const FluentProvider: ForwardRefComponent<FluentProviderProps> = React.forwardRef<
+  HTMLElement,
+  FluentProviderProps
+>((props, ref) => {
```

Comment:
explicit annotation clashes with explicit generic instantation - if explicit annotation is used generics should be avoid - that's why you were getting ts error and were forced to use `as` assertion 

```suggestion
export const FluentProvider: ForwardRefComponent<FluentProviderProps> = React.forwardRef((props, ref) => {
```